lib/bankaccounts/account.rb

```diff
@@ -0,0 +1,37 @@
+module BankAccounts
+
+  class Account
+    attr_accessor :id, :balance
+
+    def initialize(id, initial_balance)
+      # raises an error if initial_balance is negative
+      if initial_balance < 0
+        raise ArgumentError.new("Only positive initial balances allowed")
+      end
+
+      @id = id
+      @balance = initial_balance
+    end
+
+    def withdraw(amount)
+      # does not allow the account to go negative
+      # will output a warning message and return the original un-modified balance
+      if amount > @balance
+        puts "You can't withdraw what you don't have..."
+      else
+      # the input amount gets taken out of the account as result of an ATM transaction
+        @balance = @balance - amount
```

Comment:
Style tip: You can also say `@balance -= amount`, which does the same thing as line 23. Similarly, if you wanted to add something to a variable and reassign the new value to the same variable, you can do `foo += amount`